mautic/mautic
# Add support for shortener services

~~⚠️ this must be merged after https://github.com/mautic/mautic/pull/12128 and updated too.~~



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      |  https://github.com/mautic/user-documentation/pull/191
| Related developer documentation PR URL | tba
| Issue(s) addressed                     | Fixes #... 



#### Description:
For tests, you can use Bitly plugin https://github.com/Webmecanik/mautic-bitly-bundle

This PR replaces the old way to use shortener service. It was difficult to use any service with another authentication like a key in the URL.  This PR replaces it with the proper way to use the service and any integration that you want.




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Go to configuration and see Shortener service option with message about install plugin
3. Install bitly https://github.com/Webmecanik/mautic-bitly-bundle
4. Get access key from https://app.bitly.com/settings/api/ and setup/enable bitly plugin
5. Go to configuration and choose bitly as shortener service
6. Go to SMS and try send SMS with URL. You should see bitly shortener URL
